packages/react/__tests__/warnings.js

```diff
@@ -184,7 +307,7 @@ describe('unsafe pseudo classes', () => {
       expect(console.error).not.toBeCalled()
     })
 
-    test.only('does not warn when using the flag on the rule that preceeds a declaration', () => {
```

Comment:
My bad - I forgot to uncomment this and I've missed broken cases (tests added as part of this PR are new though, the crash didn't occur in the existing test suite but some expectations around those warnings were broken)